### PR TITLE
feat: add fulfill method

### PR DESCRIPTION
**Pull request by zxqfox**
cc @skad0 @tadatuta @yeti-or

### CODE REVIEW

File: lib/fulfill.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+module.exports = function (bemEntity, scope) {
+    let result = {};
```
Comment (tadatuta) [inline lib/fulfill.js]: const?

File: lib/fulfill.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        bemEntity.entity.elem || bemEntity.entity.modName &&
+            (scope.entity.elem && (result.entity.elem = scope.entity.elem));
+        bemEntity.entity.modVal && scope.entity.modName &&
+            (result.entity.modName = scope.entity.modName) && (result.entity.modVal = bemEntity.entity.modVal);
+        // if entity has only val for mod, we shoul use mod name from scope
+        (!bemEntity.entity.modName && bemEntity.entity.modVal && scope.entity.modName) &&
+            (result.entity.modName = scope.entity.modName);
+    }
+
+    result.tech = bemEntity.tech || (scope.tech && (result.tech = scope.tech));
```
Comment (Yeti-or) [inline lib/fulfill.js]: `result.tech = bemEnity.tech || scope.tech` ?

File: lib/fulfill.js (hunk 1 above)
Comment (skad0) [inline lib/fulfill.js]: Почему?

File: lib/fulfill.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+module.exports = function (bemEntity, scope) {
+    let result = {};
+
+    result.entity = {};
```
Comment (tadatuta) [inline lib/fulfill.js]: Вкусовщина:

```js
const result = {
    entity: {},
    tech: bemEntity.tech || scope.tech
};

if (Object.keys(bemEntity).length === 1 && bemEntity.tech) {
    result.entity = Object.assign({}, scope.entity);
    return result;
}

// и дальше без лишней индентации
```

File: test/fulfill.test.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+test('fulfill should support tech grabbing from scope', t => {
+    const entity = { entity: { block: 'block' } };
+    const scope = { entity: { block: 'block' }, tech: 'js' };
+
+    t.deepEqual(fulfill(entity, scope), {
+        entity: { block: 'block' }, tech: 'js'
+    });
+});
+
+test('entity tech should dominate scope', t => {
```
Comment (tadatuta) [inline test/fulfill.test.js]: BDSM: BEM dominates scope, motherf*cker!

File: lib/fulfill.js (hunk 1 above)
Comment (tadatuta) [inline lib/fulfill.js]: Оно же нигде не перетирается

File: lib/fulfill.js (hunk 3 above)
Comment (zxqfox) [inline lib/fulfill.js]: Вот не поверишь...

**Comment by zxqfox** [comment]
Need one more review, please

**Comment by tadatuta** [comment]
lgtm

**Comment by Yeti-or** [comment]
why it is called `fulfill` ?

**Comment by zxqfox** [comment]
Because it fulfils partial data with scope's entity and tech. What you suggest?

**Comment by Yeti-or** [comment]
I suggest fu**ll**fill
`fulfill/fulfil` — satisfy
`fullfill` — full fill as completely pack

or I didn't understand what method do 😃

**Comment by Yeti-or** [comment]
everything else is 🆗

**Comment by zxqfox** [comment]
Ok, renamed to `assign`. ;-)

**Pull request merged**